remotes/docker/resolver.go

```diff
@@ -82,6 +83,9 @@ type ResolverOptions struct {
 	// Host provides the hostname given a namespace.
 	Host func(string) (string, error)
 
+	// UserAgent specifies the resolver's UserAgent
+	UserAgent string
```

Comment:
This could be generalized as custom headers

Reply:
I've added a custom `Headers` field instead